test/integration/expression-tests/to-string/basic/test.json

```diff
@@ -15,16 +15,6 @@
       "isZoomConstant": true,
       "type": "String"
     },
-    "outputs": [
-      "1",
-      "false",
-      "null",
-      {
-        "error": "Expected a primitive value in [\"string\", ...], but found Array<Number, 2> instead."
-      },
-      {
-        "error": "Expected a primitive value in [\"string\", ...], but found Object instead."
-      }
-    ]
+    "outputs": ["1", "false", "null", "[1,2]", "{\"y\":1}"]
```

Comment:
I am, yeah.  It feels to me like `to-string` should always work, and since, with the exception of `Color`, all the values in our system can be represented in JSON just fine, I think it seems fine to commit to `JSON.stringify`.  (Once `Color`s can be contained within objects and arrays, we can use the `replacer` parameter of `JSON.stringify` to deal with those.)

Reply:
I don't think we want to commit to having a string representation of every type we ever add to the expression type system. What, for example, would be the result of `["to-string", ["image", "foo"]]` (https://github.com/mapbox/mapbox-gl-js/issues/5261)?

Reply:
Eh, I think it's okay to come up with some (obviously lossy) way to stringify `Image` values -- `{"image": "foo"}` or something -- but even if not, I'd still be in favor of being as liberal as possible.  If nothing else, I could imagine it being super convenient while debugging to set a symbol's text-field property to `["to-string", ["get", "offset"]]` (when we finally have support for array values in feature properties)